src/Pim/Bundle/CatalogBundle/Elasticsearch/ProductIndexer.php

```diff
@@ -88,15 +88,19 @@ public function indexAll(array $products, array $options = [])
      */
     public function remove($productId, array $options = [])
     {
-        $this->indexer->delete($this->indexType, $productId);
+        // TODO: WARNING, HERE WE HARDCORE PRODUCT MODEL INDEX TYPE, TOFIX
+//        $this->indexer->delete($this->indexType, $productId);
+        $this->indexer->delete($this->productModelIndexType, $productId);
```

Comment:
When this method is called, the product entity does not have its id anymore. So we cannot give product object to this method.